src/Battlescape/ExplosionBState.cpp

```diff
@@ -110,16 +110,20 @@ void ExplosionBState::init()
 			_damageType = itemRule->getDamageType();
 		}
 
-		if (type == BT_PSIAMP || _hit)
+		if ((type == BT_PSIAMP
+			 || (type == BT_SCRIPTED
+				  && (action == BA_MINDCONTROL || action == BA_PANIC))) || _hit)
 		{
 			Position targetPos = _center.toTile();
 			_targetPsiOrHit = _parent->getSave()->getTile(targetPos)->getOverlappingUnit(_parent->getSave());
 		}
 
 		//testing if we hit target
-		if (type == BT_PSIAMP && !_hit)
+		if ((type == BT_PSIAMP
+			 || (type == BT_SCRIPTED
+				 && (action == BA_MINDCONTROL || action == BA_PANIC))) && !_hit)
 		{
-			if (action != BA_USE)
+			if (action != BA_USE && type != BT_SCRIPTED)
```

Comment:
why is this here? what does it mean?

Reply:
It just means, that the default code path for the PSI Amp is not being used by the scripted item.